02-Generics/Generics/Generics.cs

```diff
@@ -91,9 +102,41 @@ public static class ArrayExtentions {
 		///     { 1, "a", false },
 		///   }
 		/// </example>
-		public static void SortTupleArray<T1, T2, T3>(this Tuple<T1, T2, T3>[] array, int sortedColumn, bool ascending) {
-			// TODO :SortTupleArray<T1, T2, T3>
-			// HINT : Add required constraints to generic types
+		/// 
+		private static object GetItem<T1, T2, T3>(this Tuple<T1, T2, T3> tupleObject, int index)
+		{
+			if (index == 0) return tupleObject.Item1;
+			if (index == 1) return tupleObject.Item2;
+			if (index == 2) return tupleObject.Item3;
+			return null;
+		}
+		public static void SortTupleArray<T1, T2, T3>(this Tuple<T1, T2, T3>[] array, int sortedColumn, bool ascending)
+		{
+			int Comparer<T>(T x, T y, bool asc) where T : IComparable
+			{
+				if (asc) return x.CompareTo(y); ;
+				return y.CompareTo(x);
+			}
+
+			if (sortedColumn < 0 || sortedColumn > 2)
+				throw new IndexOutOfRangeException();
+			if (array.Length <= 0)
```

Comment:
array length is always >= 0. You should be able to use this function on empty arrays